crates/bevy_ecs/src/system/system_registry.rs

```diff
@@ -172,6 +177,73 @@ impl World {
         }
         Ok(())
     }
+
+    /// Run a system one-shot by itself.
+    /// It calls like [`RunSystemOnce::run_system_once`](crate::system::RunSystemOnce::run_system_once),
+    /// but behaves like [`World::run_system`].
+    /// System will be registered on first run, then its state is kept for after calls.
+    ///
+    /// # Limitations
+    ///
+    /// Likes [`World::run_system`], with additionally:
+    ///
+    /// - Closure systems are not allowed: systems are identified by their name.
+    pub fn run_system_singleton<M, T: IntoSystem<(), (), M>>(
+        &mut self,
+        system: T,
+    ) -> Result<(), UnregisteredSystemError> {
+        // create the hashmap if it doesn't exist
+        let mut system_map = {
+            let mut map = self.get_resource_mut::<SystemMap>();
+            if map.is_none() {
+                self.insert_resource::<SystemMap>(SystemMap {
+                    map: bevy_utils::HashMap::default(),
+                });
+                map = self.get_resource_mut::<SystemMap>();
+            };
+            map.unwrap()
+        };
+
+        let system = IntoSystem::<(), (), M>::into_system(system);
+        // use the system name as the key
+        let system_name = system.name().clone();
+        // check if the system is a closure
+        if system_name.contains("{{closure}}") {
+            error!("Cannot run along a closure system");
+            return Err(UnregisteredSystemError::Closure);
+        }
+        // take system out
+        let RegisteredSystem {
+            mut initialized,
+            mut system,
+        } = system_map
+            .map
+            .remove(system_name.deref())
+            .unwrap_or(RegisteredSystem {
+                initialized: false,
+                system: Box::new(system),
+            });
```

Comment:
This always `Box`es  the system, even when it was found in the map. Consider using `unwrap_or_else`:

```suggestion
            .unwrap_or_else(|| RegisteredSystem {
                initialized: false,
                system: Box::new(system),
            });
```